paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
```

Comment:
At first, I think that if ```malloc``` method gets an error, the client can do nothing to overcome the problem. 
Second, if we check the state of ```malloc```, then we have to check the state when we construct an array. It's will be quite fussy.